pytorch_lightning/core/datamodule.py

```diff
@@ -423,6 +423,10 @@ def __new__(cls, *args: Any, **kwargs: Any) -> "LightningDataModule":
         obj.prepare_data = cls._track_data_hook_calls(obj, obj.prepare_data)
         obj.setup = cls._track_data_hook_calls(obj, obj.setup)
         obj.teardown = cls._track_data_hook_calls(obj, obj.teardown)
+
+        # calling LightningDataModule.__init__ on obj, to ensure in all cases of inheritance it is called,
+        # even if explicit call to super().__init__ hasn't been made
```

Comment:
nit: we use backticks for code in comments

```suggestion
        # calling this to ensure the `LightningDataModule` is initialized for all cases of inheritance,
        # even if `super().__init__` hasn't been explicitly called in the class
```